staging/src/k8s.io/client-go/scale/client_test.go

```diff
@@ -213,10 +246,10 @@ func fakeScaleClient(t *testing.T) (ScalesGetter, []schema.GroupResource) {
 	client := New(fakeClient, restMapper, dynamic.LegacyAPIPathResolverFunc, resolver)
 
 	groupResources := []schema.GroupResource{
-		{Group: corev1.GroupName, Resource: "replicationcontroller"},
-		{Group: extv1beta1.GroupName, Resource: "replicaset"},
-		{Group: appsv1beta2.GroupName, Resource: "deployment"},
-		{Group: "cheese.testing.k8s.io", Resource: "cheddar"},
+		{Group: corev1.GroupName, Resource: "replicationcontrollers"},
```

Comment:
Why do they have to be plural?

Reply:
hmm we construct the url path from `GroupVersionResource`, and the resource name in url path needs to be plural as described [here](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#verbs-on-resources).

In addtion, If we do not use plural form here, the api path would not match the resource paths defined here(note the resource name in paths): 
https://github.com/kubernetes/kubernetes/blob/64a67184cecfc96da53405c78c4e95d80cd68e0a/staging/src/k8s.io/client-go/scale/client_test.go#L161-L167

Because in `Patch` method, we do not negotiate with the server like `Get` or `Update` did:
https://github.com/kubernetes/kubernetes/blob/64a67184cecfc96da53405c78c4e95d80cd68e0a/staging/src/k8s.io/client-go/scale/client.go#L80-L84
the `Resource` field of `gvr` returned by `pathAndVersionFor` would be plural.

Reply:
Thanks for the explanation. The call to `c.mapper.ResourceFor` hid the problem for the `Get` and `Update` test.